test/parallel/test-console-assign-undefined.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+// Should be above require, because code in require read console
+// what we are trying to avoid
+// set should be earlier than get
+
+global.console = undefined;
+
+// Initially, the `console` variable is `undefined`, since console will be
+// lazily loaded in the getter.
+
+require('../common');
+const assert = require('assert');
+
+// global.console's getter is called
+// Since the `console` cache variable is `undefined` and therefore false-y,
+// the getter still calls NativeModule.require() and returns the object
+// obtained from it, instead of returning `undefined` as expected.
```

Comment:
Incomprehensible comment... I'd remove it.

Reply:
After reading it... I totally agree. I have no idea what it stands for.